lib/samples/apply_scheduled_updates_to_preplanned_map_area/apply_scheduled_updates_to_preplanned_map_area.dart

```diff
@@ -46,8 +46,8 @@ class _ApplyScheduledUpdatesToPreplannedMapAreaState
   // The Active mobile map package.
   MobileMapPackage? _mobileMapPackage;
   // Offline task and parameters used for updating the map package.
-  OfflineMapSyncTask? _offlineMapSyncTask;
-  OfflineMapSyncParameters? _mapSyncParameters;
+  late OfflineMapSyncTask? _offlineMapSyncTask;
+  late OfflineMapSyncParameters? _mapSyncParameters;
```

Comment:
This change looks weird to me -- as far as I know, we should be able to declare optionals without `late`, because they default to `null`. I don't see what's special about these two, compared to `_mobileMapPackage` just above.

Which linter rule drove this change? (Like, if you change it back, what is the analyzer message?)

Reply:
This is what the linter says ` Use 'late' for private members with a non-nullable type. Try making adding the modifier 'late'. • use_late_for_private_fields_and_variables`
But what you said about it being inconsistent makes sens becase this linter rule is marked "[experimental](https://dart.dev/tools/linter-rules/use_late_for_private_fields_and_variables)" so we might need to use this one with caution (or remove it?)

Reply:
Actually I think I see what it's saying. Every time we access `_offlineMapSyncTask`, we use `!`, which means we're treating it as if it is never null. So there's no point in us declaring it nullable. It makes sense that it's telling us to use `late` in that case. But it's a little off -- the underlying problem is really that we shouldn't have used `?` in the declaration. So I think we should have this:
```dart
  late OfflineMapSyncTask _offlineMapSyncTask;
  late OfflineMapSyncParameters _mapSyncParameters;
```
This change will mean we have to remove the `!` wherever we access the variable, which is reasonable. Then once those changes are made, all the lint rules should be satisfied.

Reply:
okay, I will make sure to update them!